### PR TITLE
Add products for `python`, also build shared library

### PR DESCRIPTION
**Pull request by staticfloat**
We need more snakes in our lives

### CODE REVIEW

File: P/Python/build_tarballs.jl [hunk 1]
```diff
@@ -65,6 +66,8 @@ platforms = filter(p -> !isa(p, Windows), platforms)
 
 # The products that we will ensure are always built
 products = Product[
+    ExecutableProduxct(:python, ["python", "python3"]),
```
Comment (ararslan) [inline P/Python/build_tarballs.jl]:
```suggestion
    ExecutableProduct(:python, ["python", "python3"]),
```

Comment (giordano) [reply P/Python/build_tarballs.jl]: We can't make it

**Comment by giordano** [comment]
Is there a libpython too?

**Comment by staticfloat** [comment]
Yes; it can be useful for embedding.  I think my naming is not correct though.  :P

**Comment by giordano** [comment]
Aren't the arguments swapped?

File: P/Python/build_tarballs.jl [hunk 2]
```diff
@@ -65,6 +66,8 @@ platforms = filter(p -> !isa(p, Windows), platforms)
 
 # The products that we will ensure are always built
 products = Product[
+    ExecutableProduct(["python", "python3"], :python),
+    LibraryProduct(["libPython3.8"], :libpython),
```
Comment (giordano) [inline P/Python/build_tarballs.jl]: Just libpython3 maybe?

**Pull request merged**